packages/open-next/src/http/openNextResponse.ts

```diff
@@ -174,18 +169,29 @@ export class OpenNextNodeResponse extends Transform implements ServerResponse {
         ...this.headers,
         ...this.initialHeaders,
       };
+      const initialCookies = parseCookies(
+        (this.initialHeaders[SET_COOKIE_HEADER] as string | string[]) ?? [],
+      ) as string[];
+      //Do we want to filter out the cookies that are already set?
+      // At the moment cookies from the middlewware will override the ones set in the route
+      this._cookies = [...this._cookies, ...initialCookies];
```

Comment:
Good question, we should ask the community? In terms of flow, middleware comes before the route so I'd assume the route should have the final say in the cookies. Just my pov, but you could argue the other way depending on your use case.

Reply:
Yeah we could ask the community.
Right now for headers, middleware will override route, it makes sense that it's the same for cookies. The thing with the route having the final say is that in some case ( app router for example ) there is no way to modify the headers others than by having the middleware take precedence. 

There is also another option here, we could use an env variable (or an option in the config) to decide which should take priority.